### PR TITLE
After dying, you won't respawn with your previous save.

### PR DESCRIPTION
**Pull request by Adrijaned**
Also, there goes autosaving, every 60 seconds!

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
I think I need a bit more detail on this. I know #313 is a thing and Ererbe mentioned a similar case of losing stuff on Discord. But on very latest develop I can't reproduce it. Or at least I come back with the same fancy stuff I started with rather than in a blank newbie ship.

Auto-save is a fine thing and all but I'm not sure how it relates to the dying bit? Have you been able to reproduce or sort out the related code to explain the mysteries? :-)

In other words: this change looks fine when it comes to keeping a better save. But how does it differ from current state and what *should* actually happen? Is it that you respawn with the same setup you picked in new game, so if you start with a loaded imperial you *always* start with that if you die, but we want to make it so you respawn in the exact same state that you died in? And do we want that despite it removing a certain element of risk?

**Comment by Adrijaned** [comment]
How to easily test this: on develop, start a new game ith loaded imperial
big, then exit to menu, then start a new game with imperial small,then drop
all your items, then ram into everything solid until you die,then respawn,
you'll be in the imperialBig again

čt 4. 10. 2018 v 5:04 odesílatel Rasmus Praestholm <notifications@github.com>
napsal:

**Comment by Adrijaned** [comment]
In one of oniatus' refactors of create player workflow, which accepts the new ship for player to create with items and such, or null if it is o be the same as previous, which is used on startup/respawn. However, when called with null, it loads the previous ship from savefile (y'know, startup), but for respawn, this means you respawn with ship you last used before saving.

**Comment by Cervator** [comment]
Ahh. Alright. Lets fix that part then! In addition to auto-save :-)

I'll merge this and submit a separate issue. Thanks!

**Comment by Cervator** [comment]
Merged! I switched to 30 sec auto-save to match Terasology, hope that's OK :-)

**Comment by rraallvv** [comment]
On Android Lollipop the game is not restored and the button "Previous Ship" is always disabled in the menu "New Game", does anyone know if this is a known issue?
